### PR TITLE
Update Wasm docs for new containerd shims release

### LINKED ISSUE #4343: Wasm docs need to be updated for spin v2 and lunatic
/kind cleanup

**What needs cleanup:**

**Describe the solution you'd like**

`containerd-wasm-shims` keeps moving forward, so we should update the existing docs in CAPZ to mention the newer runtimes. See kubernetes-sigs/image-builder#1350.

Files to change: 1

- [ ] https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/book/src/topics/wasi.md

**Anything else you would like to add:**

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Updates the documentation about [WebAssembly](https://capz.sigs.k8s.io/topics/wasi) to sync with newer containerd shims added in kubernetes-sigs/image-builder#1350.

**Which issue(s) this PR fixes**:

Fixes #4343

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
/hold

This should merge after image-builder v0.1.22 is out, which it isn't quite yet.

File: docs/book/src/topics/wasi.md [hunk 1]
```diff
@@ -181,7 +199,7 @@ Deploy these resources to your workload cluster:
 kubectl --kubeconfig=<workload-kubeconfig> apply -f wasm-runtimes-constrained.yaml
 ```
 
-In the preceding YAML, note the nodeSelector and the label. The Kubernetes scheduler will select nodes with the `cluster.x-k8s.io/wasmtime-slight-v1: "true"`, `cluster.x-k8s.io/wasmtime-spin-v1: "true"`, or `cluster.x-k8s.io/wasmtime-wws-v1: "true"` label to determine where to schedule Wasm workloads.
+In the preceding YAML, note the nodeSelector and the label. The Kubernetes scheduler will select nodes with the `cluster.x-k8s.io/wasmtime-slight-v1: "true"`, `cluster.x-k8s.io/wasmtime-slight-v1: "true"`, `cluster.x-k8s.io/wasmtime-spin-v2: "true"`, or `cluster.x-k8s.io/wasmtime-wws-v1: "true"` label to determine where to schedule Wasm workloads.
```
Comment (nojnhuh) [inline docs/book/src/topics/wasi.md:202]: I think this is what this should be if I'm interpreting this right?
```suggestion
In the preceding YAML, note the nodeSelector and the label. The Kubernetes scheduler will select nodes with the `cluster.x-k8s.io/wasmtime-lunatic-v1: "true"`, `cluster.x-k8s.io/wasmtime-slight-v1: "true"`, `cluster.x-k8s.io/wasmtime-spin-v2: "true"`, or `cluster.x-k8s.io/wasmtime-wws-v1: "true"` label to determine where to schedule Wasm workloads.
```

**Review by nojnhuh — COMMENTED** [review]
/lgtm

**Comment by mboersma** [comment]
Updated with a short section for "How do I write a lunatic app?" and with one reference that I had missed to spin **v2**.

File: docs/book/src/topics/wasi.md (hunk 1 above)
Comment (nojnhuh) [reply docs/book/src/topics/wasi.md:207]: Did this get lost in the last push?

Comment (mboersma) [reply docs/book/src/topics/wasi.md]: Indeed--fixed now.

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/hold cancel

**Review by Jont828 — APPROVED** [review]
/lgtm

Thanks for keeping docs up to date!

**Comment by devigned** [comment]
lgtm! Nice work, @mboersma!

**Pull request merged**